### PR TITLE
Automate publish on version changes

### PR DESCRIPTION
**Pull request by ericmj**
Uses the new package scoped API keys from Hex.

### CODE REVIEW

File: .github/workflows/outdated.yml [hunk 1]
```diff
@@ -16,8 +16,8 @@ jobs:
       - name: Install OTP and Elixir
         uses: erlef/setup-beam@v1
         with:
-          otp-version: 24.0
-          elixir-version: 1.12.2
+          otp-version: 25.2
```
Comment (whatyouhide) [inline .github/workflows/outdated.yml:19]: Nitpick: latest?

```suggestion
          otp-version: 25.3
```

File: .github/workflows/publish.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+changed_version="$(git diff --name-only HEAD~1 HEAD -- VERSION)"
+
+if [[ "${changed_version}" ]]; then
+  echo "Publishing because the version changed since the last commit"
+
+  git config user.name "${GITHUB_ACTOR}"
+  git config user.email "${GITHUB_ACTOR}@users.noreply.github.com"
+  git tag "v$(shell cat VERSION)"
```
Comment (whatyouhide) [inline .github/workflows/publish.sh:10]: Does this strip the potential newline in the `VERSION` file?

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  push:
+    branches:
+      - main
+
+jobs:
+  outdated:
+    name: Check outdated
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
```
Comment (whatyouhide) [inline .github/workflows/publish.yml:13]:
```suggestion
      - uses: actions/checkout@v3
```

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+      - name: Install OTP and Elixir
+        uses: erlef/setup-beam@v1
+        with:
+          otp-version: 25.2
```
Comment (whatyouhide) [inline .github/workflows/publish.yml:18]:
```suggestion
          otp-version: 25.3
```

**Comment by wojtekmach** [comment]
Awesome!

File: .github/workflows/publish.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+name: Publish
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  outdated:
+    name: Check outdated
```
Comment (ericmj) [inline .github/workflows/publish.yml:10]:
```suggestion
  publish:
    name: Publish
```

**Review by whatyouhide — APPROVED** [review]

File: .github/workflows/publish.sh (hunk 2 above)
Comment (ericmj) [reply .github/workflows/publish.sh]: It looks like `$(...)` strips it.

**Comment by ericmj** [comment]
I did some testing but cannot 100% test the full flow until there's a new release. Let's hope it works 🤞

**Pull request merged**